plugins/utils/scope.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, helpers
+, config
+, pkgs
+, ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "scope";
+  originalName = "scope.nvim";
+  defaultPackage = pkgs.vimPlugins.scope-nvim;
+
+  maintainers = [ lib.maintainers.insipx ];
+
+  settingsOptions = {
+    hooks = {
+      pre_tab_enter = helpers.defaultNullOpts.mkLuaFn "" ''
+        Run custom logic before entering a tab.
+      '';
```

Comment:
If we don't know the plugin default, can you use the `'` variant and _not_ supply a `pluginDefault`?

```nix
pre_tab_enter = helpers.defaultNullOpts.mkLuaFn' {
  description = ''
    Run custom logic before entering a tab.
  '';
}
```

Or alternatively, if the plugin defaults to the option being nil/undefined, perhaps we should set our `pluginDefault` to `null`?